### PR TITLE
fix: [CDS-40969]: disable save button on yaml errors

### PR DESCRIPTION
**Pull request by radujomir**
### Summary

Disable yaml save button when editor is invalid

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by yc147 — APPROVED** [review]

**Review by avinashmadhwani02 — APPROVED** [review]

**Review by mayankVermaHarness — DISMISSED** [review]

**Comment by radujomir** [comment]
trigger rebase

**Comment by radujomir** [comment]
trigger cypress

File: src/modules/30-secrets/pages/secretDetails/SecretDetails.tsx [hunk 1]
```diff
@@ -21,6 +21,7 @@ import {
   VisualYamlToggle
 } from '@wings-software/uicore'
 
+import isEmpty from 'lodash/isEmpty'
```
Comment (abhinavrastogi-harness) [inline src/modules/30-secrets/pages/secretDetails/SecretDetails.tsx:24]:
```suggestion
import { isEmpty } from 'lodash-es'
```

File: src/modules/30-secrets/pages/secretDetails/SecretDetails.tsx [hunk 2]
```diff
@@ -74,6 +74,7 @@ const YAMLSecretDetails: React.FC<YAMLSecretDetailsProps> = ({ refetch, secretDa
     ProjectPathProps & SecretsPathProps & ModulePathParams
   >()
   const [yamlHandler, setYamlHandler] = React.useState<YamlBuilderHandlerBinding | undefined>()
+  const [hasValidationErrors, setHasValidationErrors] = React.useState<boolean>(false)
```
[the remark this answers is not in the document]
Comment (abhinavrastogi-harness) [reply src/modules/30-secrets/pages/secretDetails/SecretDetails.tsx:79]: I have the same question. What was the resolution here?

File: src/modules/30-secrets/pages/secretDetails/SecretDetails.tsx [hunk 3]
```diff
@@ -192,14 +195,21 @@ const YAMLSecretDetails: React.FC<YAMLSecretDetailsProps> = ({ refetch, secretDa
     openDialog()
   }
 
+  const handleChange = React.useCallback(() => {
```
Comment (abhinavrastogi-harness) [inline src/modules/30-secrets/pages/secretDetails/SecretDetails.tsx:198]: why do we need useCallback?

File: src/modules/30-secrets/pages/secretDetails/SecretDetails.tsx [hunk 4]
```diff
@@ -192,14 +195,21 @@ const YAMLSecretDetails: React.FC<YAMLSecretDetailsProps> = ({ refetch, secretDa
...
+    requestAnimationFrame(() => {
```
Comment (abhinavrastogi-harness) [inline src/modules/30-secrets/pages/secretDetails/SecretDetails.tsx:199]: why do we need raf?
copying and validating latest yaml to state on every frame seems very costly.

**Review by abhinavrastogi-harness — CHANGES_REQUESTED** [review]

File: src/modules/30-secrets/pages/secretDetails/SecretDetails.tsx (hunk 3 above)
Comment (radujomir) [reply src/modules/30-secrets/pages/secretDetails/SecretDetails.tsx:198]: we keep the same function to be executed until yamlhandler is changed

File: src/modules/30-secrets/pages/secretDetails/SecretDetails.tsx (hunk 1 above)
Comment (radujomir) [reply src/modules/30-secrets/pages/secretDetails/SecretDetails.tsx:24]: will change

File: src/modules/30-secrets/pages/secretDetails/SecretDetails.tsx (hunk 4 above)
Comment (radujomir) [reply src/modules/30-secrets/pages/secretDetails/SecretDetails.tsx:199]: the onChange is handled until the value is updated, so we need to wait for the right state in order to perform some logic

**Review by abhinavrastogi-harness — APPROVED** [review]
depending on RAF for functionality seems hacky. that should not be needed unless you are working outside the React lifecycle.
it can bring new timing issues in the future.
Approved the PR for now.. but do look into refactoring this into a more stable and maintainable implementation.

File: src/modules/30-secrets/pages/secretDetails/SecretDetails.tsx (hunk 1 above)
Comment (abhinavrastogi-harness) [reply src/modules/30-secrets/pages/secretDetails/SecretDetails.tsx:24]: please fix this before merging.

**Comment by radujomir** [comment]
retrigger jest

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**